snapshot/search.go

```diff
@@ -51,65 +53,67 @@ func visitmimes(snap *Snapshot, opts *SearchOpts) (iter.Seq2[*vfs.Entry, error],
 		return nil, err
 	}
 
-	prefix := "/" + opts.Mime
-	if !strings.HasSuffix(prefix, "/") {
-		prefix += "/"
-	}
-
-	// XXX: to further optimize this routine we'd need a ScanGlob
-	// method.  Basically what we're trying to do here is to
-	// resolve globs like "application/*/prefix/*"
-	it, err := idx.ScanFrom(prefix)
-	if err != nil {
-		return nil, err
-	}
-
 	return func(yield func(*vfs.Entry, error) bool) {
-		for it.Next() {
-			p, mac := it.Current()
-
-			if !strings.HasPrefix(p, prefix) {
-				break
+		for _, mime := range opts.Mimes {
+			prefix := "/" + mime
+			if !strings.HasSuffix(prefix, "/") {
+				prefix += "/"
 			}
 
-			entry, err := fsp.ResolveEntry(mac)
+			// XXX: to further optimize this routine we'd need a ScanGlob
+			// method.  Basically what we're trying to do here is to
+			// resolve globs like "application/*/prefix/*"
+			it, err := idx.ScanFrom(prefix)
 			if err != nil {
 				yield(nil, err)
```

Comment:
nit: here we're missing a return afterwards, just in case the calling code don't `break` after the error.

Reply:
fixed